packages/web-components/src/components/ic-popover-menu/ic-popover-menu.tsx

```diff
@@ -100,18 +108,67 @@ export class PopoverMenu {
   }
 
   componentDidRender(): void {
-    if (this.open) {
-      createPopper(this.anchorEl, this.host, {
-        placement: "bottom-start",
-        modifiers: [
-          {
-            name: "offset",
-            options: {
-              offset: [0, 4],
+    if (this.firstRender && this.open) {
+      this.firstRender = false;
+      let adjust = false;
+
+      const dialogEl = this.host.closest("ic-dialog");
+      const onDialog = dialogEl !== null;
+
+      if (onDialog) {
+        this.host.classList.add("on-dialog");
```

Comment:
could set a flag to show it on a dialog & then conditionally apply class in the `render`

Reply:
Conditionally rendering the class in the render causes issues and never applies the on-dialog class. We do classList.add in other places so I think this should be okay to stay like this in here and ic-tooltip.